nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/metadata/RemoveMirrorTransformer.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2017-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.p2.internal.metadata;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLEventFactory;
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLEventWriter;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.Attribute;
+import javax.xml.stream.events.StartElement;
+import javax.xml.stream.events.XMLEvent;
+
+/**
+ * Find and remove the p2.mirrorsUrl property
+ *
+ * @since 1.next
+ */
+public class RemoveMirrorTransformer
```

Comment:
unit tests

Reply:
Its covered by an integration test in `P2ProxyIT.testArtifacts_mirrorRemoved` - https://github.com/sonatype-nexus-community/nexus-repository-p2/pull/64/files/cae244c34a72385ebe8158ff303522c839cb41b3#diff-f550338175e8addcad0a1b15e1897ae6R360-R370